Sources/CMakeLists.txt

```diff
@@ -49,74 +66,139 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+## ** RAKE! need to make this pandora safe.	  **
+# compiler specific flags
+if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
+	# This section and the like are for flags/defines that can be shared between 
+	# c and c++ compile options
+	add_compile_options(-Wall)
+	add_compile_options(-pipe)
+	add_compile_options(-fPIC)
+	add_compile_options(-march=native)
+	add_definitions(-D_REENTRANT=1)
+	add_definitions(-D_MT=1)
+
+	## Add your custom C and CXX flags here.
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
+	## For C flags
+	set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS} -g -D_DEBUG=1 -DDEBUG=1 -O0")
+	set(CMAKE_C_FLAGS_DEBUGALL "${CMAKE_C_FLAGS} -g -ggdb3 -D_DEBUG=1 -DDEBUG=1 -O0")
+	set(CMAKE_C_FLAGS_PROFILE "${CMAKE_C_FLAGS} -g -ggdb3 -D_DEBUG=1 -DDEBUG=1 -O0 -fno-omit-frame-pointer")
+	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS} -DNDEBUG=1 -D_NDEBUG=1 -O3 -ffast-math -fno-unsafe-math-optimizations -fomit-frame-pointer")
+	set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS} -DNDEBUG=1 -D_NDEBUG=1 -g -O3 -ffast-math -fno-unsafe-math-optimizations -fomit-frame-pointer")
+	set(CMAKE_C_FLAGS_MINSIZEREL "${CMAKE_C_FLAGS} -DNDEBUG=1 -D_NDEBUG=1 -Os -ffast-math -fno-unsafe-math-optimizations -fomit-frame-pointer")
+
+	## For C++ flags
+	set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS} -g -D_DEBUG=1 -DDEBUG=1 -O0")
+	set(CMAKE_CXX_FLAGS_DEBUGALL "${CMAKE_CXX_FLAGS} -g -ggdb3 -D_DEBUG=1 -DDEBUG=1 -O0")
+	set(CMAKE_CXX_FLAGS_PROFILE "${CMAKE_CXX_FLAGS} -g -ggdb3 -D_DEBUG=1 -DDEBUG=1 -O0 -fno-omit-frame-pointer")
+	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -DNDEBUG=1 -D_NDEBUG=1 -O3 -ffast-math -fno-unsafe-math-optimizations -fomit-frame-pointer")
+	set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS} -DNDEBUG=1 -D_NDEBUG=1 -g -O3 -ffast-math -fno-unsafe-math-optimizations -fomit-frame-pointer") ## RAKE! Does -DNDEBUG=1 and -D_NDEBUG=1 mess with RelWithDebInfo?
+	set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS} -DNDEBUG=1 -D_NDEBUG=1 -Os -ffast-math -fno-unsafe-math-optimizations -fomit-frame-pointer")
+
+
+	add_compile_options(-fno-strict-aliasing)
+
+	CHECK_CXX_COMPILER_FLAG("-fvisibility=hidden" cxx_has_fvisibility)
+	if(NOT cxx_has_fvisibility)
+		message(FATAL_ERROR "Compiler does not support -fvisibility")
+	endif()
+	#add_compile_options(-fvisibility=hidden) ## RAKE! This causes some issues with undefined symbols.
+
+	CHECK_CXX_COMPILER_FLAG("-Woverloaded-virtual" cxx_has_Woverload_virtual)
+	if(cxx_has_Woverload_virtual)
+		add_compile_options(-Woverloaded-virtual)
+	endif()
+
+	# TODO fix these warnings
+	add_compile_options(-Wno-sign-compare)
+	add_compile_options(-Wno-switch)
+	add_compile_options(-Wno-format-security)
+
+	if(MACOSX)
+		add_definitions(-DMACOS_X=1)
+
+		add_compile_options(-arch x86_64 -mmacosx-version-min=10.6)
+		set(ldflags "${ldflags} -arch x86_64 -mmacosx-version-min=10.6")
+
+		set(sys_libs ${sys_libs} "-framework Carbon -framework Cocoa -framework IOKit")
```

Comment:
It's possible we don't need to link to these frameworks now, too. We had to with SDL 1.2, because SDL_main referenced them and was statically linked to the main binary, but there's no SDL_main in SDL2).

(I'm sorry, I'm nitpicking the hell out of this patch.)